tests/math_tools/test_curl.cpp

```diff
@@ -0,0 +1,134 @@
+#include <gtest/gtest.h>
+
+#include "../coord_transformations/geometry_coord_transformations_tests.hpp"
+
+#include "circular_to_cartesian.hpp"
+#include "combined_mapping.hpp"
+#include "curl.hpp"
+#include "cylindrical_to_cartesian.hpp"
+#include "indexed_tensor.hpp"
+#include "mesh_builder.hpp"
+#include "metric_tensor_evaluator.hpp"
+#include "tensor.hpp"
+#include "toroidal_to_cylindrical.hpp"
+#include "vector_field.hpp"
+
+
+// Try to confirm that \curl (\grad psi) == 0
+// psi = a * sinh(r) * cos(theta)
+// B = \grad psi
+// \curl B = \curl (\grad psi) = 0
+TEST(CurlTests, irrotational_field_has_zero_curl)
+{
+    using Mapping2D = CircularToCartesian<Rho, Theta, R, Z>;
+    using ToroidalMapping = ToroidalToCylindrical<Mapping2D, Zeta, Phi>;
+    using CylindricalMapping = CylindricalToCartesian<R, Z, Zeta, X, Y>;
+
+    double R0 = 6.2;
+    Coord<R, Z> origin_point(R0, 0.0);
+    Mapping2D polar_to_RZ(origin_point);
+    ToroidalMapping toroidal_to_cylindrical(polar_to_RZ);
+    CylindricalMapping cylindrical_to_cartesian;
+    CombinedMapping<CylindricalMapping, ToroidalMapping, Coord<Rho, Theta, Phi>>
+            mapping(cylindrical_to_cartesian, toroidal_to_cylindrical);
+    Curl curl(mapping);
+
+    using BasisSpatial = VectorIndexSet<Rho, Theta, Phi>;
+    using CovBasisSpatial = get_covariant_dims_t<BasisSpatial>;
+    IdxStepRho nrho(10);
+    IdxStepTheta ntheta(10);
+    IdxStepPhi nphi(2);
+    ddc::init_discrete_space<GridRho>(
+            GridRho::init<GridRho>(Coord<Rho>(0.0), Coord<Rho>(1.0), nrho));
+    ddc::init_discrete_space<GridTheta>(GridTheta::init<GridTheta>(
+            build_uniform_break_points(Coord<Theta>(0.0), Coord<Theta>(2 * M_PI), ntheta)));
+    ddc::init_discrete_space<GridPhi>(
+            GridPhi::init<GridPhi>(Coord<Phi>(0.0), Coord<Phi>(2 * M_PI), nphi));
+
+    Coord<Rho, Theta, Phi> coord
+            = ddc::coordinate(Idx<GridRho, GridTheta, GridPhi>(nrho - 1, ntheta - 5, 0));
```

Comment:
Are the calls to `init_discrete_space` necessary? Couldn't you simply define a test coordinate?